examples/alice_objectbox/lib/main.dart

```diff
@@ -0,0 +1,192 @@
+import 'package:alice/alice.dart';
+import 'package:alice_http/alice_http_adapter.dart';
+import 'package:alice_http/alice_http_extensions.dart';
+import 'package:alice_objectbox/alice_objectbox.dart';
+import 'package:alice_objectbox/alice_store.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+Future<void> main() async {
+  /// This is required so ObjectBox can get the application directory
+  /// to store the database in.
+  WidgetsFlutterBinding.ensureInitialized();
+
+  /// Initialize AliceStore before running the app.
+  final AliceStore store = await AliceStore.create();
+
+  runApp(MyApp(store: store));
+}
+
+class MyApp extends StatefulWidget {
+  const MyApp({
+    super.key,
+    required this.store,
+  });
+
+  final AliceStore store;
+
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  late final AliceHttpAdapter _aliceHttpAdapter = AliceHttpAdapter();
+
+  late final Alice _alice = AliceObjectBox(
```

Comment:
I'm thinking about chaning it slightly. Instead of doing separate `AliceObjectBox` can we have maybe additional configuration parameter in Alice where we will put the `AliceStore`, so you won't need to use separate `AliceCore` instance?

And in original `Alice` package we can check whether store has been used (optionally) and wire data through that store. This will require additional abstraction in `Alice` but potentially in the future we could simply use another db without changing anything in `Alice.`  Also this will be "real" plugin architecture - user will only pass his store without changing too much of `Alice` configuration.

I can help you with that if you want :)

Reply:
```dart
  /// Create store. Store implements AliceStore class which provides common
  /// methods for store.
  final store = AliceObjectBoxStore();
  
  late final Alice _alice = Alice(
    showNotification: true,
    showInspectorOnShake: true,
    maxCallsCount: 1000,
    /// Alice knows here that it should use store and wires data through that.
    store: store
  );
```

Reply:
Well see, the problem with that is that all these various databases vary all so slightly so I don't think having this "chose your own database" is possible. After all, one has to create `objectbox.g.dart` and `objectbox-model.json` using a build runner. Mess with those files and the whole thing goes up in smoke.

Reply:
Then there's also the problem that the UI kinda relies on Dart native list sorting which is for the most part very memory inefficient. That can be all offset directly to ObjectBox's queries, however, it would require a lot of additional abstraction. 🤓

Reply:
I'll try and see if I can make some magic happen with generics etc but no promises 😅

Reply:
I gave it a shot, however, while the store can be abstracted, i.e.

```dart
abstract interface class AliceStore {
  // ... stuff
}

class AliceObjectBoxStore implements AliceStore {
  // ... stuff
}
```

and then fed to `Alice` either int he default constructor or in a custom named constructor, like say

```dart
final AliceStore? _store;
  
Alice.withStore({
  required AliceStore store,
  GlobalKey<NavigatorState>? navigatorKey,
  bool showNotification = true,
  bool showInspectorOnShake = false,
  String notificationIcon = '@mipmap/ic_launcher',
  int maxCallsCount = 1000,
  TextDirection? directionality,
  bool? showShareButton,
}) : _store = store {
  _navigatorKey = navigatorKey;
  _aliceCore = buildAliceCore();
}
```
the `AliceCore` is ObjectBox specific, in this case `AliceCoreObjectBox` and would either have to be somehow provided by the user beforehand which is crazy. That why I went down the route of simply _extending_ Alice rather than creating a _"real plugin"_. It makes much more sense, as it separates the concerns and doesn't hinder any further package development.